test/integration/api/forms/draft.js

```diff
@@ -1401,15 +1401,75 @@ describe('api: /projects/:id/forms (drafts)', () => {
               should.not.exist(body.details);
             }))));
 
-      it('should reject if the user cannot modify', testService((service) =>
-        service.login('alice', (asAlice) =>
-          asAlice.post('/v1/projects/1/forms/simple/draft')
+      const withRole = role => [
+        `has role "${role}"`,
+        (service, asAlice) => service.login('chelsea', async (asChelsea) => {
+          await asChelsea.get('/v1/users/current')
+            .expect(200)
+            .then(({ body }) => body)
+            .then((chelsea) =>
+              asAlice.post(`/v1/projects/1/assignments/${role}/${chelsea.id}`)
+                .expect(200));
+          return asChelsea;
+        }),
+      ];
+
+      const noExtraRoles = () => [
+        'does not have project-specific permissions',
+        (service) => service.login('chelsea'),
+      ];
+
+      [
+        noExtraRoles(),
+        withRole('app-user'),
+        withRole('formfill'),
+        withRole('formview'),
+        withRole('pub-link'),
+        withRole('viewer'),
```

Comment:
I've updated the test setup to make it clear that all of these roles should have identical access to a form draft as a user with no project-specific permissions.